guides/common/modules/proc_configuring-satellite-with-an-alternate-cname.adoc

```diff
@@ -0,0 +1,16 @@
+[id='configuring-satellite-with-an-alternate-cname_{context}']
+= Configuring {Project} with an Alternate CNAME
+Use this procedure to configure {Project} with an alternate CNAME. Note that the procedures for users of a default {Project} certificate and custom certificate differ.
+
+.For Default {Project} Certificate Users
+
+If you use a default {Project} certificate and want to configure {Project} with an alternate CNAME, enter the following command on {Project} to include a new CNAME into the {Project} certificate:
+
+[options="nowrap" subs="+quotes,attributes"]
+----
+{foreman-installer} --certs-cname _alternate_fqdn_
```

Comment:
Looking at comments in https://bugzilla.redhat.com/show_bug.cgi?id=1160344 , this only works during initial installation of Satellite - you can't change, or specify alternate, cname on already installed server.

But it seems we never mention that requirement explicitly. Moreover, placement of this section suggests this is something that can be freely done as post-install configuration. That tripped me up when I tried to follow these instructions.

Reply:
I think you *can* do this after the fact, but then need to pass one of the --regenerate-certs (can't remember the exact name) parameters, as the installer otherwise won't touch existing certs.

This comes at a cost of probably invalidating client configurations, tho.

Reply:
True, the option is `--certs-reset`. With that, I could get new certs and complete all instructions.

Do we expect people to just know that?

Reply:
Given the implications, I would neither expect people to know, nor document it to be used here.

The normal user workflow should be "do this during initial installation"

Reply:
We try to separate different scenarios into different procedures to not end up having a large procedure just on installing with many conditional notes in it. I have written a procedure for those who have already installed Satellite, and informed them that they need to configure it with an alternate CNAME before they register any host or Capsule to avoid complications with certs. In this procedure, I used the `--certs-reset` option. I also created another conditional bullet for those who have not installed Satellite yet to say that they can install it with an alternate cname right away,

Reply:
I understand this might be seen as more complex, and there are many conditionals, but I do like this new version much better. We are clear that alternative CNAME is something you should think about before installing Satellite. If you already installed Satellite, we give working command - but warn that there will be additional work on clients and capsules. And we also give exact command to use when Satellite is installed for the first time.

Overall, great job!